src/pineko/cli/scaffold.py

```diff
@@ -0,0 +1,50 @@
+"""'scaffold' mode of CLI."""
+import pathlib
+
+import click
+import rich
+
+from .. import configs, scaffold
+from ._base import command
+
+
+@command.group("scaffold")
+@click.option(
+    "-c",
+    "--configs",
+    "cfg",
+    default=None,
+    type=click.Path(resolve_path=True, path_type=pathlib.Path),
+    help="Explicitly specify config file (it has to be a valid TOML file).",
+)
+def scaffold_(cfg):
+    """Manage folders needed for the project as spelled out in the configuration file."""
+    path = configs.detect(cfg)
+    base_configs = configs.load(path)
+    configs.configs = configs.defaults(base_configs)
```

Comment:
Ah wait, the problem *is* here, since this code is run always, also when the user is just asking for help.

I propose this should be solved in this PR, at least for this subcommand.

Reply:
Yes but, as I was saying in the other comment, this is exactly the same of the `theory` command so, whatever solution we want to apply here, we need to copy it also there.

Reply:
Yep, that is fine, but at least we close this PR with a working command. Then we open a new one for fixing all the others.
However, if it becomes urgent we can release without (but at the moment is not terribly urgent)

Reply:
No, I agree to solve it here Anyway I am not sure how. We for sure want the parent command to set the configs. Do we want to use a `try, except`? Or do you have any suggestions?

Reply:
I'm looking for a solution, but I do not have an optimal one. The best would have been that it were already doing on its own, as it is for the parent command, but I don't know why it isn't...
Have a quick look online, I'm doing as well. In any case, if possible we should try to detect an help request (i.e. when `click` would decide to print help or a prompt for missing arguments/wrong CLI invocation) and avoid execution at all, but otherwise any working solution might be fine for the time being.

Reply:
https://github.com/pallets/click/issues/295

Reply:
Ok, it is complicated: here there is a rather generic (and lengthy) solution https://github.com/pallets/click/issues/295#issuecomment-611648993

I'd try to keep it smaller, but I'm failing for the time being:
```py
import click

CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])


@click.group(context_settings=CONTEXT_SETTINGS)
@click.option("-h", "--help", is_flag=True)
def command(help):
    """Base command"""
    print("This should not bee seen if asking for help.")


@command.command()
@click.argument("ciao")
def sub(ciao):
    """Subcommand"""
    print(f"Ciao: {ciao} (but not during help)")
```

While this is working:
```py
import sys

import click

CONTEXT = {"help_option_names": ["-h", "--help"]}


def print_cmd_help(ctx, param, value):
    group = ctx.command
    # parse the cmdline and get the command and its arguments
    parser = group.make_parser(ctx)
    global_opts, args, _ = parser.parse_args(args=sys.argv[1:])
    if global_opts.get("help") is True or not args:
        # global help wanted
        click.echo(ctx.get_help())
        ctx.exit()

    # create the command object manually and parse its arguments
    name, cmd, args = group.resolve_command(ctx, args)
    help_names = cmd.get_help_option_names(ctx)

    if set(args) & set(help_names) or not args:
        # return command help when a user wants it or when he does not provide
        # arguments
        # taken from https://github.com/pallets/click/blob/7.1.1/src/click/core.py#L597
        cmd_ctx = click.Context(cmd, info_name=cmd.name, parent=ctx)
        click.echo(cmd_ctx.get_help())
        cmd_ctx.exit()


@click.group(context_settings=CONTEXT)
@click.option(
    "-h",
    "--help",
    is_flag=True,
    callback=print_cmd_help,
    expose_value=False,
    is_eager=True,
)
def command():
    print("This should not bee seen if asking for help.")


@command.command(help="this is the help of test")
@click.argument("ciao")
def sub(ciao):
    print(f"Ciao: {ciao} (but not during help)")
```

Reply:
I propose to take the second one, put the function in a `library.py` module, and add there an alias to the full decorator as well, such that we can simply add `@library.help` to groups.

We should also improve a bit, since e.g. the `Context` construction has been updated in `click.core` as well https://github.com/pallets/click/blob/5e75fe7f9c1b820d71f482599cd6996523332d06/src/click/core.py#L744

Reply:
Other pain point is that I'm not sure that `1` in `parser.parse_args(args=sys.argv[1:])` is always correct (you could have a nested subgroup).
At the moment I'd just leave a `TODO:` comment there, since I don't know how to detect how much nested is the current command (it might be also just fine, if `click` manipulates the `sys.argv` list, and the `1` thus is the correct choice...)

Reply:
Further pain point is if we have a subcommand that should actually be invoked without any argument, in which case the second `not args` check would be wrong.

Reply:
Yes, for instance `scaffold new` is called without arguments and it is exaclty one of the commands raising the error.

Reply:
Moved into its own issue #78